packages/react-native/android/src/main/java/com/dashx/sdk/DashXActivityLifecycleCallbacks.kt

```diff
@@ -0,0 +1,91 @@
+package com.dashx.sdk
+
+import android.app.Activity
+import android.app.Application
+import android.content.Context
+import android.content.pm.ActivityInfo
+import android.content.pm.PackageManager
+import android.os.Bundle
+import com.facebook.react.bridge.Arguments
+
+class DashXActivityLifecycleCallbacks : Application.ActivityLifecycleCallbacks {
+    private val dashXClient = DashXClient.instance
+    private var startSession = System.currentTimeMillis().toDouble()
+
+    init {
+        dashXClient.trackAppStarted()
+    }
+
+    override fun onActivityPaused(activity: Activity?) {
+        if (!lifecycleTrackingEnabled) {
+            return
+        }
+
+        dashXClient.trackAppSession(System.currentTimeMillis().toDouble() - startSession)
+    }
+
+    override fun onActivityResumed(activity: Activity?) {
+        if (!lifecycleTrackingEnabled) {
+            return
+        }
+
+        startSession = System.currentTimeMillis().toDouble()
+    }
+
+    override fun onActivityStarted(activity: Activity?) {
+        if (!screenTrackingEnabled) {
+            return
+        }
+
+        val packageManager = activity?.packageManager
+        val info = packageManager?.getActivityInfo(activity.componentName, PackageManager.GET_META_DATA)
+        val activityLabel = info?.loadLabel(packageManager)
+        activityLabel?.let { it -> dashXClient.screen(it.toString(), Arguments.createMap()) }
```

Comment:
Double checking: Doesn’t activity.onStart() get called when the device orientation changes?

Reply:
No